### PR TITLE
related to issue:#8077

### PR DESCRIPTION
**Pull request by imiskolee**
allow special interval parse for unit:hour(same as unit:day)

### What problem does this PR solve? 

1. use interval unit day special parse logic to handle interval unit hour format.
2. add more test cases.
more details:(https://github.com/pingcap/tidb/issues/8077)

### Check List 

Tests 

 - Unit test

Code changes

- Just change function inside

Side effects

- Nothing

Related changes

- Nothing

### CODE REVIEW

**Comment by zz-jason** [comment]
@imiskolee Please merge master and resolve conflicts.

**Comment by imiskolee** [comment]
@zz-jason  already fix on master branch. please review again.  many thanks!

File: expression/builtin_time.go [hunk 1]
```diff
@@ -2553,8 +2553,8 @@ func (du *baseDateArithmitical) getIntervalFromString(ctx sessionctx.Context, ar
 	if isNull || err != nil {
 		return "", true, errors.Trace(err)
 	}
-	// unit "DAY" has to be specially handled.
-	if strings.ToLower(unit) == "day" {
+	// unit "DAY" and "HOUR" has to be specially handled.
+	if strings.ToLower(unit) == "day" || strings.ToLower(unit) == "hour" {
```
Comment (ngaut) [inline expression/builtin_time.go]: We can just call ToLower once.

Comment (imiskolee) [reply expression/builtin_time.go]: it's mean we need a temp variable for ToLower?

Comment (imiskolee) [reply expression/builtin_time.go]: @ngaut already remove redundant ToLower call.

**Comment by ngaut** [comment]
Please sign the cla.

**Pull request closed without merging**